app/views/layouts/_navigation.html.erb

```diff
@@ -12,9 +12,9 @@
     <ul class="nav navbar-nav navbar-right">
       <form class="navbar-form navbar-left" role="search" class="search_form" style="margin-top:6px;margin-bottom:4px;">
         <div class="form-group">
-          <input name ="q" type="text" class="form-control input-sm" placeholder="Пошук" size="30"/>
+          <%= text_field_tag 'q', nil, size: 30, class: 'form-control input-sm', value: params[:q] %>
```

Comment:
1. Use symbol as param name: `text_field_tag :q`
2. Second param is the value. No need to set it explicitly with `value: params[:q]`, just use `text_field_tag :q, params[:q], size: 30, class: 'form-control input-sm'`